augur/application/schema/alembic/versions/18_schedule_any_old_facade_repositories_to.py

```diff
@@ -0,0 +1,77 @@
+"""Schedule any old facade repositories to be re-cloned
+
+Revision ID: 18
+Revises: 17
+Create Date: 2023-05-02 17:35:18.891913
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.sql import text
+import pathlib
+import shutil
+from augur.application.config import AugurConfig
+from augur.application.db.session import DatabaseSession
+import logging
+
+# revision identifiers, used by Alembic.
+revision = '18'
+down_revision = '17'
+branch_labels = None
+depends_on = None
+
+logger = logging.getLogger(__name__)
+
+
+def total_facade_reset():
+    conn = op.get_bind()
+
+    conn.execute(text(f"""
+
+    UPDATE augur_operations.collection_status
+    SET facade_status='Pending', facade_task_id=NULL, facade_weight=NULL,commit_sum=NULL,facade_data_last_collected=NULL;
+
```

Comment:
I think we should only run this on repos that don't have a commit_count or have a repos that are stored in the old structure. The reason being that I would like this script to reset all of facade on a huge instane if someone accidentally downgrades the database

Reply:
If they wanted to downgrade the schema it would be proper to also reset the facade repos again. We also can't check based on the commit_count because the revision before this one has commit_count without the directory change. I think it's fine how it is in my opinion.

Reply:
I don't think it matters if someone accidentally clears their facade repos while downgrading the schema since it's so fast to clone.

Reply:
Reply to first part: How would it be proper? I don't see why that would be the desired behavior. Usually in my experience downgrades are to bring the database back to the prior state, but resetting the repos doesn't do anything to reset to the old state. 

Reply to second part: I think we can do it based on commit_count we would run it if the facade directory was wrong or the commit_count doesn't exist since they are both valid times to do it.

I do think it matters, because although it is fast to clone, each repo still has to go through the collecting phase and if there are 22,000 repos that go reset and there collections finished instantly (since the commits exist) it would take around 6 hours for all the repos to get back through to success and this is not even considering the time to add all the tasks to the queue, and the time to clone.

Reply:
> I do think it matters, because although it is fast to clone, each repo still has to go through the collecting phase and if there are 22,000 repos that go reset and there collections finished instantly (since the commits exist) it would take around 6 hours for all the repos to get back through to success and this is not even considering the time to add all the tasks to the queue, and the time to clone.

The only issue with this is that it sets all the repos to new but otherwise it doesn't really change the underlying behavior of facade. I do agree that people could see that as a problem though since people want to prioritize the newly added repos.

Reply:
I approved because it does solve the problem. I'm just a bit apprehensive about doing this without conditions because it may not do what people would expect it to do when it is run multiple times

Reply:
> I approved because it does solve the problem. I'm just a bit apprehensive about doing this without conditions because it may not do what people would expect it to do when it is run multiple times

I see the concern but I think any issue this alembic script might cause would be more of an inconvenience caused by not knowing facade repos are all being deleted. If we want to we can add a confirm prompt so that people are absolutely aware that their facade directories are being deleted. If we want to we can also mark the old repos or something so that they aren't re-collected as new.